### PR TITLE
[120] Handle connection state changes in Publisher and Subscriber.

### PR DESCRIPTION
**Pull request by robertkuraj**

### CODE REVIEW

**Review by KacperKluka — CHANGES_REQUESTED** [review]
There are a few things to fix but overall it looks good :wink: 
Also I've discussed some things with @robertkuraj on a call so he'll be fixing the logic and places for checking if the state has changed.

File: Subscriber/Sources/Extensions/ARTRealtimeChannelState+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Ably
+
+/**
+ Public wrapper for ARTRealtimeChannelState, as we don't want to pass it to our clients
+ */
+
+extension ARTRealtimeChannelState {
```
[the remark this answers is not in the document]
Comment (robertkuraj) [reply Subscriber/Sources/Extensions/ARTRealtimeChannelState+Extensions.swift]: Updated.

File: Subscriber/Sources/Extensions/ARTRealtimeConnectionState+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Foundation
+import Ably
+
+/**
+ Public wrapper for ARTRealtimeConnectionState, as we don't want to pass it to our clients
+ */
+
+extension ARTRealtimeConnectionState {
```
[the remark this answers is not in the document]
Comment (robertkuraj) [reply Subscriber/Sources/Extensions/ARTRealtimeConnectionState+Extensions.swift]: Updated.

File: Subscriber/Sources/Services/AblyPublisherPresence.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        switch self {
+        case .enter, .present, .update:
+            return .online
+        default:
+            return .offline
+        }
+    }
+}
+
+extension ARTPresenceAction {
```
[the remark this answers is not in the document]
Comment (robertkuraj) [reply Subscriber/Sources/Services/AblyPublisherPresence.swift]: Updated.

File: Subscriber/Sources/DefaultSubscriber.swift [hunk 4]
```diff
@@ -138,10 +146,60 @@ extension DefaultSubscriber {
         }
     }
     
+    private func performPresenceUpdated(_ event: PresenceUpdateEvent) {
+        switch event.presence {
+        case .enter, .present:
+            isOnline = true
```
[the remark this answers is not in the document]
Comment (robertkuraj) [reply Subscriber/Sources/DefaultSubscriber.swift]: Updated.

File: Subscriber/Sources/DefaultSubscriber.swift [hunk 5]
```diff
@@ -138,10 +146,60 @@ extension DefaultSubscriber {
...
+        default:
+            isOnline = false
```
[the remark this answers is not in the document]
Comment (robertkuraj) [reply Subscriber/Sources/DefaultSubscriber.swift]: Updated.
https://github.com/ably/ably-asset-tracking-cocoa/pull/128/commits/d8396a3dc507daf6850551fec3ff7a5b0a928f31

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Pull request merged**